allennlp/data/dataset_readers/dataset_reader.py

```diff
@@ -84,8 +85,14 @@ class DatasetReader(Registrable):
         parameters you set for this DatasetReader!_
     """
 
-    def __init__(self, lazy: bool = False, cache_directory: str = None) -> None:
+    def __init__(
+        self,
+        lazy: bool = False,
+        cache_directory: Optional[str] = None,
+        max_instances: Optional[int] = None,
```

Comment:
Can you add this to the docstring?

Reply:
bb753f02638cabaf03a207272d49647179fd6ba4